fmc/src/error.rs

```diff
@@ -0,0 +1,52 @@
+/*++
+
+Licensed under the Apache-2.0 license.
+
+File Name:
+
+    error.rs
+
+Abstract:
+
+    File contains API and macros used by the FMC for error handling
+
+--*/
+
+/// Caliptra Component
+pub enum FmcComponent {
+    /// Global Error
+    Global = 0x201,
+}
+
+#[macro_export]
+macro_rules! fmc_err_def {
+    ($comp_name:ident, $enum_name: ident { $($field_name: ident = $field_val: literal,)* }) => {
+
+        #[derive(Debug, Copy, Clone, Eq, PartialEq)]
+        #[allow(clippy::enum_variant_names)]
+        pub enum $enum_name {
+            $($field_name = $field_val,)*
+        }
+
+        impl From<$enum_name> for u32 {
+            fn from(val: $enum_name) -> Self {
+                ((($crate::error::FmcComponent::$comp_name) as Self) << 24) | (val as Self)
```

Comment:
Doesn't `0x201 << 24` overflow a 32-bit number, making this 0x02000000, which colides with `0x101 << 24` in the ROM? @vsonims?

Reply:
I think we need to have a discussion about create a canonical registry of error codes. Why not just have a central crate with ALL the caliptra error codes, that can be referenced from test cases and caliptra clients? (including a Debug impl when compiled with the std feature)

Reply:
I can move this code to common. Let me tweak that.

Reply:
I trimmed the PR to just detect panic - we need to propose a collision-free partition for error namespace with Vishal and make it common.